src/test/java/com/google/graphgeckos/dashboard/buildbot/BuildBotClientTest.java

```diff
@@ -0,0 +1,266 @@
+     * Responds with the original (taken from the API) clang-x86_64-debian-fast JSON.
+     */
+    private final MockResponse CLANG_VALID_RESPONSE = new MockResponse()
+      .setResponseCode(200)
+      .setHeader(
+        HttpHeaders.CONTENT_TYPE,
+        MediaType.APPLICATION_JSON_VALUE)
+      .setBody(BUILD_BOT_CLANG.getContent());
+
+    /**
+     * Response when the {@code VALID_BUILD_BOT_NAME_FUCHSIA} data is requested. Status OK (200),
+     * Responds with the original (taken from the API) fuchsia-x86_64-linux JSON.
+     */
+    private final MockResponse FUCHSIA_VALID_RESPONSE = new MockResponse()
+      .setResponseCode(200)
+      .setHeader(
+        HttpHeaders.CONTENT_TYPE,
+        MediaType.APPLICATION_JSON_VALUE)
+      .setBody(BUILD_BOT_FUCHSIA.getContent());
+
+    /**
+     * Response when the {@code VALID_BUILD_BOT_NAME_FUCHSIA} data is requested. Status NOT FOUND (404).
+     */
+    private final MockResponse PAGE_NOT_FOUND_RESPONSE = new MockResponse()
+      .setResponseCode(404)
+      .setHeader(
+        HttpHeaders.CONTENT_TYPE,
+        MediaType.APPLICATION_JSON_VALUE);
+
+    /**
+     * Response when the {@code VALID_BUILD_BOT_NAME_FUCHSIA} data is requested. Status OK (200).
+     * Responds with empty JSON.
+     */
+    private final MockResponse EMPTY_JSON_RESPONSE = new MockResponse()
+      .setResponseCode(200)
+      .setHeader(
+        HttpHeaders.CONTENT_TYPE,
+        MediaType.APPLICATION_JSON_VALUE)
+      .setBody(EMPTY_JSON);
+
+    @Override
+    public MockResponse dispatch(RecordedRequest request) {
+
+      /* Expected form of the request url: baseUrl/buildBot/builds/buildId?as_text=1 . */
+      if (request.getPath().contains(VALID_BUILD_BOT_NAME_CLANG)) {
+        if (request.getPath().contains(Long.toString(INITIAL_BUILD_ID))) {
+          return CLANG_VALID_RESPONSE;
+        }
+        return FUCHSIA_VALID_RESPONSE;
+      }
+
+      if (request.getPath().contains(NOT_FOUND_BUILD_BOT_NAME)) {
+        return PAGE_NOT_FOUND_RESPONSE;
+      }
+
+      if (request.getPath().contains(EMPTY_JSON_BUILD_BOT_NAME)) {
+        return EMPTY_JSON_RESPONSE;
+      }
+
+      /*
+      Handles requests to the defined Build Bots only to verify that
+      the client performs requests of the expected form.
+      */
+      throw new IllegalStateException("Unknown build bot name");
+    }
+
+  };
+
+  /**
+   * Compares objects, stubbed from the {@link DatastoreRepository::updateRevisionEntry},
+   * with the corresponding expected values.
+   */
+  static class BuildBotDataMatcher implements ArgumentMatcher<BuildBotData> {
+
+    /**
+     * Expected output fields. See {@link com.google.graphgeckos.dashboard.datatypes.BuildBotData} to learn more.
+     */
+    private String commitHash;
+    private Timestamp timestamp;
+    private String name;
+    private BuilderStatus status;
+    private List<Log> logs;
+
+    public BuildBotDataMatcher(BuildBotClientTestInfo expected) {
+      super();
```

Comment:
I believe you don't need to call the default `super` constructor; this is done automatically.

Reply:
Removed, thanks!